presto-main/src/main/java/io/prestosql/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1993,12 +1995,33 @@ public void analyzeWhere(Node node, Scope scope, Expression predicate)
         private Scope analyzeFrom(QuerySpecification node, Optional<Scope> scope)
         {
             if (node.getFrom().isPresent()) {
+                validateFrom(node.getFrom().get(), new HashSet<>());
```

Comment:
This needs to be guarded with a feature toggle, cause it may cause behavioral regressions for users (previously running queries can fail now)